nathanielmillard/rancid-tomatillos
# Nd comments

#### What’s this PR do?

- This PR allows a user to make comments on movies.
- A user must be logged in to make a comment for a movie.
- A user can see all comments for a movie regardless of whether or not they are logged in or logged out.
- Makes 2 new directories for Comment component and Comments component, as well as their associated styles and initial testing files.
- Adds basic styling for a user to be able to see the form and leave a comment on the MovieShowPage component.
- Modifies the styling on MovieShowPage to ensure that the form to leave comments and comments themselves are properly aligned below the MovieShowPage information and graphics.

#### Where should the reviewer start?

- Git pull this branch.
- Ensure that you also have the rancid-tomatillos-microserver [located here](https://github.com/nathanielmillard/rancid-tomatillos-microserver).

#### How should this be manually tested?

- If the microserver has never been installed, be sure to clone down the link above for the microserver inside a new directory that is different from the rancid-tomatillos main directory.
- Navigate inside microserver director and run `npm install`.
- Start the server with `npm run dev` which should be loaded on `localhost:3001`.
- Navigate back inside the main director for rancid-tomatillos, run `npm install`.
- Navigate to the MovieShowPage to see a conditional render about the movie having no comments.
- Navigate to the MovieShowPage for 'The Owners' and see the 2 comments at the bottom, with a conditional render to login to comment on this movie.
- Login with our user `email: lucy@turing.io, password: password1`.
- Navigate to any movie page and ensure that a form to leave comments for that movie now shows.
- Try to leave a comment for that movie, record any error messages on submit.

#### Any background context you want to provide?

- This PR does not contain testing as this will be done in another branch.
- When a user posts a comment, it currently displays their ID as opposed to their name. After collaborating with Nathaniel and learning that he had to change many props for his iteration 7B, I decided to hold off changing the props for now to see how he had to change props and will implement this portion after that.

#### What are the relevant tickets?
- closes #47
- closes #46

#### Screenshots (if appropriate):
- N/A

#### Questions:
- N/A
